### PR TITLE
Play UI get rid

### PR DESCRIPTION
**Pull request by vinnicombe**
# Pull Request Checklist
- Add/update Welsh translation if required.
- Add any logging needed for Bridge Team monitoring.
- Add unit/acceptance/performance tests for new features.
- Make sure code coverage has not dropped.
- Run the [cysp-acceptance-test](https://github.com/hmrc/cysp-acceptance-test).
- Run the [nisp-sapeak-performance-tests](https://github.com/hmrc/nisp-sapeak-performance-tests).
- Perform relevant accessibility checks.
- Test manually in browser if required.
- Make any environment configuration changes/checks if needed.

### CODE REVIEW

File: app/uk/gov/hmrc/nisp/connectors/CitizenDetailsConnector.scala [hunk 1]
```diff
@@ -30,40 +28,41 @@ import play.api.http.Status.BAD_GATEWAY
...
-      }
+    } map {
+      case Right(response) =>
+        Right(response.json.as[CitizenDetailsResponse])
+      case Left(error) =>
+        metricsService.citizenDetailsFailedCounter.inc()
+        Left(error)
+    } recover {
+      case error: HttpException =>
+        Left(UpstreamErrorResponse(error.message, BAD_GATEWAY))
```
Comment (pascal22p) [inline app/uk/gov/hmrc/nisp/connectors/CitizenDetailsConnector.scala:62]: citizenDetailsFailedCounter.inc() is missing

File: app/uk/gov/hmrc/nisp/connectors/CitizenDetailsConnector.scala [hunk 2]
```diff
@@ -30,40 +28,41 @@ import play.api.http.Status.BAD_GATEWAY
...
-      .recover {
-        case error: HttpException =>
-          Left(UpstreamErrorResponse(error.message, BAD_GATEWAY))
-        case error =>
-          metricsService.citizenDetailsFailedCounter.inc()
-          throw error
-      }
+    } map {
+      case Right(response) =>
+        Right(response.json.as[CitizenDetailsResponse])
```
Comment (pascal22p) [inline app/uk/gov/hmrc/nisp/connectors/CitizenDetailsConnector.scala:56]: No success counter increment?

**Review by derek-gray — DISMISSED** [review]

File: app/uk/gov/hmrc/nisp/controllers/auth/AuthAction.scala [hunk 3]
```diff
@@ -63,22 +62,21 @@ class AuthActionImpl @Inject() (
           and Retrievals.trustedHelper
       ) {
 
-        case Some(nino) ~ _ ~ Some(CredentialStrength.weak) ~ _ ~ _ ~ Enrolments(_) ~ _ =>
+        case Some(_) ~ _ ~ Some(CredentialStrength.weak) ~ _ ~ _ ~ Enrolments(_) ~ _ =>
           upliftCredentialStrength
```
Comment (pascal22p) [inline app/uk/gov/hmrc/nisp/controllers/auth/AuthAction.scala:66]: Where is the CL uplift?

File: app/uk/gov/hmrc/nisp/connectors/CitizenDetailsConnector.scala (hunk 2 above)
Comment (vinnicombe) [reply app/uk/gov/hmrc/nisp/connectors/CitizenDetailsConnector.scala:56]: I haven't changed this, just refactored to make more readable

File: app/uk/gov/hmrc/nisp/connectors/CitizenDetailsConnector.scala (hunk 1 above)
Comment (vinnicombe) [reply app/uk/gov/hmrc/nisp/connectors/CitizenDetailsConnector.scala:62]: I haven't changed this, just refactored to make more readable

File: app/uk/gov/hmrc/nisp/controllers/pertax/PertaxHelper.scala [hunk 4]
```diff
@@ -26,32 +27,46 @@ import uk.gov.hmrc.http.HeaderCarrier
...
+        keystoreResult match {
+          case Some(isPertax) =>
+            metricsService.keystoreHitCounter.inc()
+            isPertax
+          case None =>
+            metricsService.keystoreMissCounter.inc()
+            false
+        }
+    } recover {
+      case _ =>
```
Comment (pascal22p) [inline app/uk/gov/hmrc/nisp/controllers/pertax/PertaxHelper.scala:67]: Logging needed?

Comment (vinnicombe) [reply app/uk/gov/hmrc/nisp/controllers/pertax/PertaxHelper.scala:67]: This ticket is to remove the LocalTemplateRenderer and FormPartialRetriever and Play UI and other older UI assets, this knocked the coverage down, which meant I had to add tests, which meant I refactored this file to make it more readable. Functionality should be as was.

File: app/uk/gov/hmrc/nisp/views/formatting/NispMoney.scala [hunk 5]
```diff
@@ -17,8 +17,31 @@
 package uk.gov.hmrc.nisp.views.formatting
 
 import play.twirl.api.Html
-import uk.gov.hmrc.play.views.formatting.Money
+
+import scala.math.BigDecimal.RoundingMode._
 
 object NispMoney {
-  def pounds(value: BigDecimal): Html = Money.pounds(value, if (value.isValidInt) 0 else 2)
+  def pounds(
```
Comment (pascal22p) [inline app/uk/gov/hmrc/nisp/views/formatting/NispMoney.scala:24]: FYI. This is what we have in tax summaries.

For useability we have:
https://github.com/hmrc/tax-summaries-frontend/blob/main/app/views/includes/CurrencyAmountView.scala.html
I think if I remember well it switch the position of the minus and pound sign for reading. So it read "minus 300 pounds" but it displays -£300

File: app/uk/gov/hmrc/nisp/connectors/CitizenDetailsConnector.scala (hunk 1 above)
Comment (pascal22p) [reply app/uk/gov/hmrc/nisp/connectors/CitizenDetailsConnector.scala:62]: Grafana dashboard does not display this metric anyway...

File: app/uk/gov/hmrc/nisp/views/formatting/NispMoney.scala (hunk 5 above)
Comment (vinnicombe) [reply app/uk/gov/hmrc/nisp/views/formatting/NispMoney.scala:24]: I just lifted and shifted the functionality from Play UI so it was a like for like change, so I could get rid of that dependency. I will raise a ticket to use play-frontend-hmrc currency UI, if it exists.

Comment (pascal22p) [reply app/uk/gov/hmrc/nisp/views/formatting/NispMoney.scala:24]: A currency element in play-frontend-hmrc would be wonderful. Ping me if you find anything :)

**Review by Tateni93 — APPROVED** [review]

**Pull request merged**